src/UI/Art/Animation/ResupplyAnimation.java

```diff
@@ -88,29 +76,26 @@ public boolean animate(Graphics g)
       double percent = (double) animTime / signUpBegin;
       int width = (int) (signWidth * percent);
       int height = (int) (signHeight * percent);
-      XYCoord signTopLeft = new XYCoord(tileCenterX - width / 2, tileCenterY - height / 2);
 
-      drawSign(g, signTopLeft.xCoord, signTopLeft.yCoord, width, height);
+      SpriteUIUtils.drawTextFrame(g, MENUBGCOLOR, MENUFRAMECOLOR, "", mapLocation.xCoord, mapLocation.yCoord, width / 2,
+          2 * drawScale + height - signHeight);
```

Comment:
This. This is stupid clever. Wouldn't it be easier to just have `SpriteUIUtils.drawTextFrame` detect that there is no text, and not add that back in?

Reply:
Also, I feel like `drawTextFrame` shouldn't need to know anything about the map. I want to just give it the image coordinates, or possibly have it return a `BufferedImage` with the sign for me to draw where I want. This would also allow it to be used in non-map-centric contexts.

Reply:
Heh. UI code draws out my hacky instincts.
Looks like your PR fixed all this jazz up, though.